### PR TITLE
improved data facet and month dropdowns

### PR DESCRIPTION
**Pull request by gordonleacock**

### CODE REVIEW

File: app/assets/javascripts/static.js [hunk 1]
```diff
@@ -3,31 +3,23 @@
...
-        $("div.search span#now-searching").addClass("hidden");
-        $("body").css("cursor", "auto");
-    // }
-
+    $("div.search span#not-searching").removeClass("hidden");
+    $("div.search span#now-searching").addClass("hidden");
+    $("body").css("cursor", "auto");
+    
+    set_dates("div.search ");
+    event.preventDefault();
```
Comment (respinos) [inline app/assets/javascripts/static.js]: You don't seem to have an `event` here.

File: app/assets/javascripts/static.js [hunk 2]
```diff
@@ -3,31 +3,23 @@
...
 
-  // Replaces the magnifying glass in the search button with a rotating wait icon
-  // See changes in customers.css.scss too, as well as /catalog and /home html.
-  $(document).on("click", 'button#search', function(){
 
+  $(document).on("click", 'div.search form button#search', function(){
```
Comment (respinos) [inline app/assets/javascripts/static.js]: Do we have other places where there are multiple `button#search` ? The earlier version was specific enough.

File: app/assets/javascripts/static.js [hunk 3]
```diff
@@ -37,93 +29,291 @@ $( document ).ready(function() {
...
+    }
+
+    // show search in progress
+    $("body").css("cursor", "wait");
+    set_dates("form.range_date_issued_yyyymmdd_ti ");
 
-    //alert("range start field is: " + $("input#range_start").val() + " and end field is: " + $("input#range_end").val());  
+    // send click event to search button at the top of the page
+    $('div.search form button#search').trigger("click");
+    event.preventDefault();
```
Comment (respinos) [inline app/assets/javascripts/static.js]: `event` needs to be a parameter in the earlier `function()` declaration.

File: app/assets/javascripts/static.js (hunk 1 above)
Comment (gordonleacock) [inline app/assets/javascripts/static.js]: I'm thinking I'd add event as a parameter like this  $( document ).ready(function(event) {

File: app/assets/javascripts/static.js (hunk 2 above)
Comment (gordonleacock) [inline app/assets/javascripts/static.js]: Yep, you're right. I was thinking of renaming the facet button to button#search when I added that. But in the end I didn't rename it.

File: app/assets/javascripts/static.js (hunk 3 above)
Comment (gordonleacock) [inline app/assets/javascripts/static.js]: I've added it to document.ready, so I think this is good. Okay?

File: app/assets/javascripts/static.js (hunk 1 above)
Comment (respinos) [inline app/assets/javascripts/static.js]: The ready() function does _not_ get an `event` object. Typically you're invoking `preventDefault` to stop the event from cascading --- there's no event in play here.

File: app/assets/javascripts/static.js (hunk 3 above)
Comment (respinos) [inline app/assets/javascripts/static.js]: Nope. `event` has to go with the **event handler**.

File: app/assets/javascripts/static.js (hunk 1 above)
Comment (gordonleacock) [inline app/assets/javascripts/static.js]: Okay, I've modified the code and will resubmit once I check that things are still working correctly.

File: app/assets/javascripts/static.js [hunk 4]
```diff
@@ -37,93 +28,292 @@ $( document ).ready(function() {
...
+    set_dates("div.search ");
+    event.preventDefault();
+
+  });
+
+  $("div.search form input, div.search form select").on("load", function(event){
     
-    set_dates();
+    set_dates("div.search ");
     event.preventDefault();
```
Comment (respinos) [inline app/assets/javascripts/static.js]: Do these elements get a `load` event? I'm not sure you need this function handler.

Comment (gordonleacock) [inline app/assets/javascripts/static.js]: Probably not, I'll remove 'em.

Comment (gordonleacock) [inline app/assets/javascripts/static.js]: Done.

**Pull request merged**